src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCoursesController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.RoatpCourseManagement.Application.Standards.Queries.GetAllProviderCourses;
+using SFA.DAS.RoatpCourseManagement.Application.Standards.Queries.GetAvailableCoursesForProvider;
+using SFA.DAS.RoatpCourseManagement.Application.Standards.Queries.GetProviderCourse;
+using System;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.RoatpCourseManagement.Api.Controllers
+{
+    [ApiController]
+    public class ProviderCoursesController : ControllerBase
+    {
+        private readonly ILogger<ProviderCoursesController> _logger;
+        private readonly IMediator _mediator;
+
+        public ProviderCoursesController(ILogger<ProviderCoursesController> logger, IMediator mediator)
+        {
+            _logger = logger;
+            _mediator = mediator;
+        }
+
+        [HttpGet]
+        [Route("providers/{ukprn}/courses/{larsCode}")]
+        public async Task<IActionResult> GetProviderCourse([FromRoute] int ukprn, [FromRoute] int larsCode)
+        {
+            if (ukprn <= 9999999)
+            {
+                _logger.LogWarning("Invalid ukprn {ukprn}", ukprn);
+                return BadRequest();
+            }
+
+            if (larsCode <= 0)
+            {
+                _logger.LogWarning("Invalid lars code {larsCode}", larsCode);
+                return BadRequest();
+            }
+
+            var providerCourseResult = await _mediator.Send(new GetProviderCourseQuery(ukprn, larsCode));
+
+            if (providerCourseResult == null)
+            {
+                _logger.LogError($"Provider Course not found for ukprn {ukprn} and lars code {larsCode}");
+                return NotFound();
+            }
+
+            return Ok(providerCourseResult);
+        }
+
+        [HttpGet]
+        [Route("providers/{ukprn}/courses")]
+        public async Task<IActionResult> GetAllProviderCourses([FromRoute] int ukprn)
+        {
+            if (ukprn <= 0)
```

Comment:
This should use same criteria as in GetProviderCourse ie  if (ukprn <= 9999999) and test coverage should reflect this